src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -108,38 +108,45 @@ public void Start()
             stopping = false;
             pipelineInvocationTasks = new ConcurrentDictionary<Task, Task>();
 
-            
-            for(var i = 0; i < numberOfClients; i++)
+            var exceptions = new ConcurrentQueue<Exception>();
+            Parallel.For(0, numberOfClients, i =>
             {
-
-                var internalReceiver = clientEntities.Get(fullPath, entity.Namespace.Alias);
-
-                if (internalReceiver == null)
+                try
                 {
-                    throw new Exception($"MessageReceiverNotifier did not get a MessageReceiver instance for entity path {fullPath}, this is probably due to a misconfiguration of the topology");
-                }
+                    var internalReceiver = clientEntities.Get(fullPath, entity.Namespace.Alias);
 
-                Func<BrokeredMessage, Task> callback = message =>
-                {
-                    var processTask = ProcessMessageAsync(internalReceiver, message);
-                    pipelineInvocationTasks.TryAdd(processTask, processTask);
-                    processTask.ContinueWith(t =>
+                    if (internalReceiver == null)
                     {
-                        Task toBeRemoved;
-                        pipelineInvocationTasks.TryRemove(t, out toBeRemoved);
-                    }, TaskContinuationOptions.ExecuteSynchronously);
-                    return processTask;
-                };
-
-                isRunning = true;
+                        throw new Exception($"MessageReceiverNotifier did not get a MessageReceiver instance for entity path {fullPath}, this is probably due to a misconfiguration of the topology");
```

Comment:
I have to admit that reading this message it's more confusing than helping. 

> this is probably due to a misconfiguration of the topology

Error message should either help to understand what exactly the problem is or how to fix it. Don't think this error message does either well.